reactive/kubernetes_master.py

```diff
@@ -168,16 +168,45 @@ def channel_changed():
     set_upgrade_needed()
 
 
+# Returns True if a is subnet of b
+# This method is copied from cpython as it is available only from
+# python 3.7
+# https://github.com/python/cpython/blob/3.7/Lib/ipaddress.py#L1000
+def is_subnet_of(a, b):
+    try:
+        # Always false if one is v4 and the other is v6.
+        if a._version != b._version:
+            raise TypeError("{} and {} are not of the same version".format(
+                a, b))
+        return (b.network_address <= a.network_address and
+                b.broadcast_address >= a.broadcast_address)
+    except AttributeError:
+        raise TypeError("Unable to test subnet containment "
+                        "between {} and {}".format(a, b))
+
+
+def is_service_cidr_expansion():
+    service_cidr_from_db = db.get('kubernetes-master.service-cidr')
+    if service_cidr_from_db:
+        current_service_cidr = ipaddress.ip_network(service_cidr_from_db)
+        new_service_cidr = ipaddress.ip_network(hookenv.config('service-cidr'))
+        if not is_subnet_of(current_service_cidr, new_service_cidr):
+            hookenv.log("WARN: New k8s service cidr not superset of old one")
+            return False
+
+    return True
+
+
 def service_cidr():
     ''' Return the charm's service-cidr config '''
-    frozen_cidr = db.get('kubernetes-master.service-cidr')
-    return frozen_cidr or hookenv.config('service-cidr')
+    return hookenv.config('service-cidr')
```

Comment:
This change seems unsafe and unnecessary.  I think this should be reverted and `freeze_service_cidr` changed as mentioned below.